doc/rules/tooManyJoins.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Too many joins
+id: too-many-joins
+---
+
+Verifies that the number of joins in SQL queries does not exceed a threshold.
+
+### Rule logic
+
+Counts the number of joins in each SQL query. If the count exceeds the configured threshold, a
+finding is reported.
+
+### Notes
+
+Queries with too many joins often have poor or unpredictable performance.
+
+### Resolution
+
+Rewrite the query to use fewer joins, or break the query into more than one query.
```

Comment:
Having multiple joins is often legitimate, and breaking the query just to satisfy some rule is not something I'm comfortable suggesting at all (it'll almost certainly make things works due to multiple database round-trips, less optimizer knowledge and more data). I feel we should instead indicate that this is more of an advisory rule for flagging queries which require a close look (especially since with DAO they can arise unintended).